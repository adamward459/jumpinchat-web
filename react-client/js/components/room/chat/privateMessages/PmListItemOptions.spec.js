/* global jest, expect, describe, it, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';
import PmListItemOptions from './PmListItemOptions.react';

describe('<PmListItemOptions />', () => {
  let props;
  beforeEach(() => {
    props = {
      menuOpen: null,
      handle: 'foo',
      userListId: 'userid',
      onOpenMenu: jest.fn(),
      handleClickOutside: jest.fn(),
      closeConversation: jest.fn(),
    };
  });

  it('should not show the menu if menuOpen is false', () => {
    const wrapper = shallow(<PmListItemOptions {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should show the menu if menuOpen is true', () => {
    props.menuOpen = 'userid';
    const wrapper = shallow(<PmListItemOptions {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
