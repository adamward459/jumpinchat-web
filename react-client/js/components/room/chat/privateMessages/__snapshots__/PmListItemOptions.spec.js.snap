// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PmListItemOptions /> should not show the menu if menuOpen is false 1`] = `
<TetherComponent
  attachment="top center"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "scrollParent",
      },
      Object {
        "attachment": "together",
        "pin": true,
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
>
  <button
    className="userList__Action"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-ellipsis-h"
    />
  </button>
</TetherComponent>
`;

exports[`<PmListItemOptions /> should show the menu if menuOpen is true 1`] = `
<TetherComponent
  attachment="top center"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "scrollParent",
      },
      Object {
        "attachment": "together",
        "pin": true,
        "to": "window",
      },
    ]
  }
  renderElementTag="div"
  renderElementTo={null}
>
  <button
    className="userList__Action"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-ellipsis-h"
    />
  </button>
  <clickOutside(ListItems)
    onClickOutside={[MockFunction]}
    options={
      Array [
        Object {
          "element": "div",
          "props": Object {
            "className": "dropdown__Option dropdown__Option-header",
          },
          "text": "Conversation with foo",
        },
        Object {
          "element": "button",
          "props": Object {
            "onClick": [Function],
          },
          "text": "Close conversation",
        },
      ]
    }
  />
</TetherComponent>
`;
